docs/user/unattended/storage.md

```diff
@@ -0,0 +1,956 @@
+
+It is also possible to specify "current" as a minimum or maximum size limit for partitions and
+logical volumes that already exist in the system (so "current" can only be used for device
+specifications that contain a `search` section). The usage of "current" and how it affects
+resizing the corresponding devices is explained at a separate section below.
+
+If the size is completely omitted for a device that already exists (ie. combined with `search`),
+then Agama would act as if both min and max limits would have been set to "current" (which implies
+the partition or logical volume will not be resized).
+
+Moreover, if the size is omitted for a device that will be created, Agama will determine the size
+limits when possible. There are basically two kinds of situations in which that automatic size
+calculation can be performed.
+
+On the one hand, the device may directly contain a `filesystem` entry specifying a mount point.
+Agama will then use the settings of the product to set the size limits.
+
+On the other hand, the size limits of some devices can be omitted if they can be inferred from other
+related devices following some rules.
+
+ - For an MD RAID defined on top of new partitions, it is possible to specify the size of all the
+   partitions that will become members of the RAID but is also possible to specify the desired size
+   for the resulting MD RAID and then the size limits of each partition will be automatically
+   inferred with a small margin of error of a few MiBs.
+ - Something similar happens with a partition that acts as the **only** physical volume of a new LVM
+   volume group. Specifying the sizes of the logical volumes could be enough, the size limits of the
+   underlying partition will match the necessary values to make the logical volumes fit. In this
+   case the calculated partition size is fully accurate.
+ - The two previous scenarios can be combined. For a new MD RAID that acts as the **only** physical
+   volume of a new LVM volume group, the sizes of the logical volumes can be used to precisely
+   determine what should be the size of the MD and, based on that, what should be the almost
+   exact size of the underlying new partitions defined to act as members of the RAID.
+
+The two described mechanisms to automatically determine size limits can be combined. Even creating
+a configuration with no explicit sizes at all like the following example.
+
+```json
+"storage": {
+    "drives": [
+        {
+            "partitions": [
+                { "alias": "pv" }
+              ]
+        }
+    ],
+    "volumeGroups": [
+        {
+            "name": "system",
+            "physicalVolumes": [ "pv" ],
+            "logicalVolumes": [
+                { "filesystem": { "path": "/" } },
+                { "filesystem": { "path": "swap" } }
+            ]
+        }
+    ]
+}
+```
+
+Assuming the product configuration specifies a root filesystem with a minimum size of 5 GiB and a
```

Comment:
What "product configuration" means? OK, I know what it means... but what about a user that knows nothing about Agama? Adding a footnote should be enough.

Reply:
I added a sentence to clarify that in a previous paragraph.